pytest_plugins/rerun_rp/rerun_user.py

```diff
@@ -0,0 +1,66 @@
+import logging
+
+from robottelo.config import settings
+from robottelo.report_portal.portal import ReportPortal
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _get_user_tests(launch, user):
+    """Returns test names from Report Portal Launch based on given user"""
+    tests = launch.tests(user=user).keys()
+    # Formating test names for 'member of pytest test items' operation
+    tests = [str(test).replace('::', '.') for test in tests]
+    return tests
+
+
+def _get_test_collection(selectable_tests, items):
+    """Returns the selected and deselected items"""
+    # Select test item if its given users tests else deselect
+    LOGGER.debug('Selecting/Deselecting tests based on user and latest launch test results..')
+    selected = []
+    deselected = []
+    for item in items:
+        test_item = f"{item.location[0]}.{item.location[2]}"
+        if test_item in selectable_tests:
+            selected.append(item)
+        else:
+            deselected.append(item)
+    return selected, deselected
+
+
+def pytest_addoption(parser):
+    """Add options for pytest to collect user specific tests"""
+    uhelp_script = '''
+        Collects user tests from report portal.
+
+        Usage: --user [value]
+
+        Value: [ Report_portal_username ]
+
+        Defect_types:
+            Collect tests of specific user from Report Portal
+    '''
+    parser.addoption("--user", nargs='?', help=uhelp_script)
+
+
+def pytest_collection_modifyitems(items, config):
+    """
+    Collects tests based on pytest option to select tests marked as failed/skipped on Report Portal
+    """
+    user_args = config.getvalue('user')
+    if not user_args:
+        return
+    rp = ReportPortal()
```

Comment:
I think these plugins should be combined in a single module/hook, otherwise we're having to do the RP launch look-ups twice, and parse through the items twice. Can we not filter out test status and user within the same hook?

Reply:
Done!